### PR TITLE
Refactor turret branch

### PR DESCRIPTION
**Pull request by maxh65000**
Also fixes some SSH key issues with config and improves overall code logging

### CODE REVIEW

**Review by Barrelwolf38087 — APPROVED** [review]
We should eventually move to WPILib's new [PIDSubsystem](https://first.wpi.edu/FRC/roborio/release/docs/java/edu/wpi/first/wpilibj2/command/PIDSubsystem.html#field.summary). The old one (used here) is deprecated.

File: src/main/java/frc/robot/Robot.java [hunk 1]
```diff
@@ -51,32 +51,24 @@
   //Variables from State
 
   
-  static Potentiometer sPot;
-  static PotPID sPotHandler;
+  Turret turretInstance;
```
Comment (Team138Entropy) [inline src/main/java/frc/robot/Robot.java]: private Turret mTurret

File: src/main/java/frc/robot/subsystems/Turret.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  private WPI_TalonSRX mTurretTalon;
+  private Potentiometer mPot;
+
+  // the target position (on a scale from 0 to 100)
+  private double mManualTargetPos = 50;
+
+  /**
+   * Set up our talon, logger and potentiometer
+   */
+  public Turret() {
```
Comment (Team138Entropy) [inline src/main/java/frc/robot/subsystems/Turret.java]: This is not a singleton. All subsystems should be singletons.

File: src/main/java/frc/robot/subsystems/Turret.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    // limit the output to prevent the motor from going too fast
+    output = Math.min(output, Config.getInstance().getDouble(Key.OI__VISION__PID__MAX_SPEED));
+    mTurretLogger.verbose("pid out " + output);
+    mTurretTalon.set(ControlMode.PercentOutput, output);
+  }
+
+  /**
+   * @return the raw POT value
+   */
+  public double getPotValue(){
```
Comment (Team138Entropy) [inline src/main/java/frc/robot/subsystems/Turret.java:83]: Probably doesn't need to be public

**Review by Team138Entropy — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/Turret.java (hunk 2 above)
Comment (maxh65000) [reply src/main/java/frc/robot/subsystems/Turret.java]: done

**Review by Team138Entropy — APPROVED** [review]

**Pull request merged**